x/incentives/keeper/group_test.go

```diff
@@ -0,0 +1,671 @@
+			// - Can only be run by an incentives module account
+			// - synching volume is not attempted (initialized to zero)
+			name:                "CreateGroupAsIncentivesModuleAcc",
+			systemUnderTestType: CreateGroupAsIncentivesModuleAcc,
+		},
+	} {
+		tc := tc
+		s.Run(tc.name, func() {
+			// We setup test state once and reuse it for all test cases
+			s.SetupTest()
+			s.Ctx = s.Ctx.WithBlockTime(defaultTime)
+
+			// Create 4 pools of each possible type
+			poolInfo := s.PrepareAllSupportedPools()
+
+			expectedGroupGaugeId := s.App.IncentivesKeeper.GetLastGaugeID(s.Ctx) + 1
+
+			// Initialize expected gauge records
+			var (
+				concentratedGaugeRecord = withRecordGaugeId(defaultZeroWeightGaugeRecord, poolInfo.ConcentratedGaugeID)
+				balancerGaugeRecord     = withRecordGaugeId(defaultZeroWeightGaugeRecord, poolInfo.BalancerGaugeID)
+				stableSwapGaugeRecord   = withRecordGaugeId(defaultZeroWeightGaugeRecord, poolInfo.StableSwapGaugeID)
+			)
+
+			// Set a custom creation fee to avoid test balances having false positives
+			// due to having OSMO added during test setup
+			s.App.IncentivesKeeper.SetParam(s.Ctx, types.KeyGroupCreationFee, customGroupCreationFee)
+
+			// Fund fee once to a specific test account
+			s.FundAcc(s.TestAccs[oneTimeFeeFundedIndex], customGroupCreationFee)
+
+			communityPoolAddress := s.App.AccountKeeper.GetModuleAddress(distrtypes.ModuleName)
+			originalCommunityPoolBalance := s.App.BankKeeper.GetAllBalances(s.Ctx, communityPoolAddress)
+			expectedCommunityPoolFeeBalance := sdk.NewCoins()
+
+			if tc.systemUnderTestType == CreateGroup {
+				s.runCreateGroupTests(poolInfo, concentratedGaugeRecord, balancerGaugeRecord, stableSwapGaugeRecord, expectedCommunityPoolFeeBalance, expectedGroupGaugeId, originalCommunityPoolBalance)
+				differentExecutionTypeCount++
+			} else if tc.systemUnderTestType == CreateGroupInternal {
+				s.runCreateGroupInternalTests(poolInfo, concentratedGaugeRecord, balancerGaugeRecord, stableSwapGaugeRecord, expectedCommunityPoolFeeBalance, expectedGroupGaugeId, originalCommunityPoolBalance)
+				differentExecutionTypeCount++
+			} else if tc.systemUnderTestType == CreateGroupAsIncentivesModuleAcc {
+				s.runCreateGroupAsIncentivesModuleAccTests(poolInfo, concentratedGaugeRecord, balancerGaugeRecord, expectedCommunityPoolFeeBalance, expectedGroupGaugeId, originalCommunityPoolBalance)
+				differentExecutionTypeCount++
+			} else {
+				s.FailNow("unknown system under test type")
+			}
+		})
+	}
+	// Ensure that we ran all the tests to prevent false positions
+	s.Require().Equal(int(TotalTypes), differentExecutionTypeCount)
+}
+
+func (s *KeeperTestSuite) runCreateGroupTests(poolInfo apptesting.SupportedPoolAndGaugeInfo, concentratedGaugeRecord types.InternalGaugeRecord, balancerGaugeRecord types.InternalGaugeRecord, stableSwapGaugeRecord types.InternalGaugeRecord, expectedCommunityPoolFeeBalance sdk.Coins, expectedGroupGaugeId uint64, originalCommunityPoolBalance sdk.Coins) {
```

Comment:
Note: most of the logic in this method is the same as the old `TestCreateGroup`. The original was refactored for reuse with other group creator method tests